src/test/java/com/microsoft/sqlserver/jdbc/bulkCopy/bulkCopyTest.java

```diff
@@ -0,0 +1,181 @@
+	{
+		DBStatement srcStmt = con.createStatement();
+		DBStatement dstStmt = con.createStatement();
+		DBResultSet srcResultSet = srcStmt.executeQuery("SELECT * FROM "+sourceTable.getEscapedTableName()+";");
+		DBResultSet dstResultSet = dstStmt.executeQuery("SELECT * FROM "+destinationTable.getEscapedTableName()+";");
+        ResultSetMetaData destMeta = ((ResultSet)dstResultSet.product()).getMetaData();
+        int totalColumns = destMeta.getColumnCount();
+        
+        // verify data from sourceType and resultSet
+        while(srcResultSet.next() && dstResultSet.next())
+        for(int i=1;i<=totalColumns;i++)
+        {
+        	//TODO: check row and column count in both the tables
+        	
+        	Object srcValue, dstValue;
+        	srcValue = srcResultSet.getObject(i);
+    		dstValue = dstResultSet.getObject(i);
+    		// Bulkcopy doesn't guarantee order of insertion -
+    		// if we need to test several rows either use primary key or validate result based on sql JOIN
+        	switch(destMeta.getColumnType(i))
+        	{
+        	case java.sql.Types.BIGINT:
+        		assertTrue((((Long) srcValue).longValue() == ((Long)dstValue).longValue()), "Unexpected bigint value");        		
+        		break;
+        	
+        	case java.sql.Types.INTEGER:
+        		assertTrue((((Integer) srcValue).intValue() == ((Integer)dstValue).intValue()), "Unexpected int value");
+        		break;
+        	
+        	case java.sql.Types.SMALLINT:
+        	case java.sql.Types.TINYINT:
+        		assertTrue((((Short) srcValue).shortValue() == ((Short)dstValue).shortValue()), "Unexpected smallint/tinyint value");
+        		break;
+        		
+        	case java.sql.Types.BIT:
+        		assertTrue((((Boolean) srcValue).booleanValue() == ((Boolean) dstValue).booleanValue()), "Unexpected bit value");
+        		break;
+        	
+        	case java.sql.Types.DECIMAL:
+        	case java.sql.Types.NUMERIC:
+        		assertTrue(0 == (((BigDecimal) srcValue).compareTo((BigDecimal) dstValue)) , "Unexpected decimal/numeric/money/smallmoney value");
+        		break;
+        		
+        	case java.sql.Types.DOUBLE:
+        		assertTrue(
+        				(((Double) srcValue).doubleValue() == ((Double) dstValue).doubleValue()), "Unexpected float value");
+        		break;
+        	
+        	case java.sql.Types.REAL:
+        		assertTrue((((Float) srcValue).floatValue() == ((Float) dstValue).floatValue()), "Unexpected real value");
+        		break;
+        	
+        	case java.sql.Types.VARCHAR:
+        	case java.sql.Types.NVARCHAR:
+        		assertTrue(((((String) srcValue).trim()).equals(((String) dstValue).trim())), "Unexpected varchar/nvarchar value ");
+        		break;
+        		
+        	case java.sql.Types.CHAR:
+        	case java.sql.Types.NCHAR:
+        		assertTrue((((String) srcValue).equals(((String) dstValue))), "Unexpected char/nchar value ");
```

Comment:
for char and nchar the trailing space is preserved, so I din't add it. Now that you mention it, do you think they needed for nvarchar and varchar?